TGeant3/TGeant3TGeo.cxx

```diff
@@ -2179,21 +2179,30 @@ void TGeant3TGeo::SetSensitiveDetector(const TString &volumeName, TVirtualMCSens
       tgeoExtension = it->second;
    }
 
-   // Set extension to TGeoVolume with given name,
-   // print error message if volume does not exist or
-   // if the volume already has an extension
-   TGeoVolume *volume = gGeoManager->FindVolumeFast(volumeName);
-   if (!volume) {
-      Error("SetSensitiveDetector", "volume %s not found. Setting was ignored.\n", volumeName.Data());
-   } else {
-      // Issue a warning if the volume already has another extension
-      if (volume->GetUserExtension()) {
-         Warning("SetSensitiveDetector", "The existing extension of volume %s will be lost.\n", volumeName.Data());
-      }
-
-      // attach the extension to the volume
-      volume->SetUserExtension(tgeoExtension);
+   //
+   // Set extension to any instance of the TGeoVolume with the specified name.
+   // Extension holds the registered SD object.
+   //
+   //   Error if the named volume is not found.
+   //   Error if we overwrite a user's extension.
+   //
+   TIter next( gGeoManager->GetListOfVolumes() );
+   TGeoVolume* volume = 0;
+   while ( (volume = dynamic_cast<TGeoVolume*>( next() )) ) {
```

Comment:
There is no reason to use `dynamic_cast,` as the volume list always contains TGeoVolume type; this can be changes in `static_cast`

Reply:
If a user adds something other than a volume to the list of volumes, dynamic cast would safely catch it.  But I suppose it would cause big problems elsewhere, so static should be fine here.